src/raqm.c

```diff
@@ -1123,38 +1215,82 @@ _raqm_itemize (raqm_t *rq)
   }
 
   if (rq->base_dir == RAQM_DIRECTION_RTL)
-    par_type = FRIBIDI_PAR_RTL;
+    par_type = RAQM_PAR_RTL;
   else if (rq->base_dir == RAQM_DIRECTION_LTR)
-    par_type = FRIBIDI_PAR_LTR;
+    par_type = RAQM_PAR_LTR;
 
   if (rq->base_dir == RAQM_DIRECTION_TTB)
   {
     /* Treat every thing as LTR in vertical text */
     max_level = 1;
-    memset (types, FRIBIDI_TYPE_LTR, rq->text_len);
+    memset (types, RAQM_TYPE_LTR, rq->text_len);
     memset (levels, 0, rq->text_len);
     rq->resolved_dir = RAQM_DIRECTION_LTR;
   }
   else
   {
+#ifdef RAQM_SHEENBIDI
+    SBCodepointSequence codepointSequence =
+    {
+      SBStringEncodingUTF8,
+      (void *) rq->text,
+      rq->text_len,
+    };
+
+    /* paragraph */
+    bidiAlgorithm =
+      SBAlgorithmCreate(&codepointSequence);
+
+    firstParagraph =
+      SBAlgorithmCreateParagraph(
+        bidiAlgorithm,
+        0,
+        INT32_MAX,
+        SBLevelDefaultLTR);
```

Comment:
Thais should take raqm par dir into account, unconditionally using `SBLevelDefaultLTR` is probably wrong.

Reply:
Fixed in 74616ea070178739adbad929edb1818355811c1c, also fixes the issue witth `max_level`.
Now only 14 tests pass, and 32 fail.